test/identity_cache_test.rb

```diff
@@ -23,4 +27,16 @@ def test_should_use_cache_in_transaction
       assert_equal false, IdentityCache.should_use_cache?
     end
   end
+
+  def test_should_use_cache_inside_transaction_on_specific_model
+    ModelWithConnection.transaction do
+      assert_equal false, IdentityCache.should_use_cache?
+    end
+  end
+
+  def test_should_use_cache_inside_unjoinable_transaction
+    ActiveRecord::Base.transaction(joinable: false) do
+      assert_equal true, IdentityCache.should_use_cache?
+    end
+  end
```

Comment:
`IdentityCache::TestCase` doesn't include `ActiveRecord::TestFixtures`, so `use_transactional_fixtures` isn't available and none of the tests are wrapped in transactions.

My intention here was that this new test would cover the transactional fixtures case by manually opening an unjoinable transaction, and [this existing test](https://github.com/Shopify/identity_cache/blob/v0.3.2/test/identity_cache_test.rb#L17-L19) would cover the case where no transactions are open and a `NullTransaction` is returned.

However the intention of the tests is clearer when `use_transactional_fixtures` is used, so I've changed the tests to use different test case classes that include `ActiveRecord::TestFixtures` and set `use_transactional_fixtures`.